src/deluge/memory/general_memory_allocator.cpp

```diff
@@ -159,12 +158,12 @@ void* GeneralMemoryAllocator::alloc(uint32_t requiredSize, bool mayUseOnChipRam,
 }
 
 uint32_t GeneralMemoryAllocator::getAllocatedSize(void* address) {
-	uint32_t* header = (uint32_t*)((uint32_t)address - 4);
-	return (*header & SPACE_SIZE_MASK);
+	uintptr_t header = (uintptr_t)address - 4;
```

Comment:
This still needs to be a pointer - it's a size_t * (pointer to a size_t) since it's the address of where the allocation size is stored